src/middlewared/middlewared/utils/nginx.py

```diff
@@ -21,21 +21,22 @@ def get_remote_addr_port(request):
         except (KeyError, ValueError):
             pass
         else:
-            if process := get_peer_process(remote_addr, remote_port):
-                if process.name() == "nginx":
-                    try:
-                        with open("/var/run/nginx.pid") as f:
-                            nginx_pid = int(f.read().strip())
-                    except Exception:
-                        pass
-                    else:
-                        try:
-                            ppid = process.ppid()
-                        except psutil.ProcessNotFound:
-                            pass
-                        else:
-                            if ppid == nginx_pid:
-                                remote_addr = x_real_remote_addr
-                                remote_port = x_real_remote_port
+            try:
+                with open("/var/run/nginx.pid") as f:
+                    nginx_pid = int(f.read().strip())
+            except Exception:
+                pass
+            else:
+                try:
+                    process = psutil.Process(nginx_pid)
+                except psutil.ProcessNotFound:
+                    pass
+                else:
+                    if process.name() == "nginx":
```

Comment:
wonder if using the context manager oneshot() would provide any speed boost
```
with process.oneshot():
    if process.name() == "nginx":
         blah
```

Reply:
@yocalebo `children` is not listed as a method that will benefit from `oneshot` https://psutil.readthedocs.io/en/latest/#psutil.Process.oneshot

Reply:
Right but name() is.

Reply:
What do you propose to group `name()` with to benefit from `oneshot()`?

Reply:
Just making an observation. Not asking for anything to be changed.